File: packages/python/ailabtools/image.py
from .http import Requester
from .utils import map_params
from .generated.endpoints import (
    ImageQueryingAsyncTaskResultsParams,
    ImageQueryingAsyncTaskResultsResponse,
    ImageQueryingAsyncTaskResultsParamMap,
    validate_imageQueryingAsyncTaskResults_params,
    ImageAiImageExtenderParams,
    ImageAiImageExtenderResponse,
    ImageAiImageExtenderParamMap,
    validate_imageAiImageExtender_params,
    ImageAiObjectReplacerParams,
    ImageAiObjectReplacerResponse,
    ImageAiObjectReplacerParamMap,
    validate_imageAiObjectReplacer_params,
    ImageAIImageCroppingParams,
    ImageAIImageCroppingResponse,
    ImageAIImageCroppingParamMap,
    validate_imageAIImageCropping_params,
    ImageErasureParams,
    ImageErasureResponse,
    ImageErasureParamMap,
    validate_imageErasure_params,
    ImageInvisibleImageWatermarkParams,
    ImageInvisibleImageWatermarkResponse,
    ImageInvisibleImageWatermarkParamMap,
    validate_imageInvisibleImageWatermark_params,
    ImageInvisibleTextWatermarkParams,
    ImageInvisibleTextWatermarkResponse,
    ImageInvisibleTextWatermarkParamMap,
    validate_imageInvisibleTextWatermark_params,
    ImageIntelligentCompositionParams,
    ImageIntelligentCompositionResponse,
    ImageIntelligentCompositionParamMap,
    validate_imageIntelligentComposition_params,
    ImagePhotoEditingParams,
    ImagePhotoEditingResponse,
    ImagePhotoEditingParamMap,
    validate_imagePhotoEditing_params,
    ImageRemoveObjectsParams,
    ImageRemoveObjectsResponse,
    ImageRemoveObjectsParamMap,
    validate_imageRemoveObjects_params,
    ImageRemoveObjectsAdvancedParams,
    ImageRemoveObjectsAdvancedResponse,
    ImageRemoveObjectsAdvancedParamMap,
    validate_imageRemoveObjectsAdvanced_params,
    ImageRemoveObjectsProParams,
    ImageRemoveObjectsProResponse,
    ImageRemoveObjectsProParamMap,
    validate_imageRemoveObjectsPro_params,
    ImageAICartoonGeneratorParams,
    ImageAICartoonGeneratorResponse,
    ImageAICartoonGeneratorParamMap,
    validate_imageAICartoonGenerator_params,
    ImageColoringParams,
    ImageColoringResponse,
    ImageColoringParamMap,
    validate_imageColoring_params,
    ImageStyleTransferParams,
    ImageStyleTransferResponse,
    ImageStyleTransferParamMap,
    validate_imageStyleTransfer_params,
    ImageStyleMigrationParams,
    ImageStyleMigrationResponse,
    ImageStyleMigrationParamMap,
    validate_imageStyleMigration_params,
    ImageColorEnhancementParams,
    ImageColorEnhancementResponse,
    ImageColorEnhancementParamMap,
    validate_imageColorEnhancement_params,
    ImageContrastEnhancementParams,
    ImageContrastEnhancementResponse,
    ImageContrastEnhancementParamMap,
    validate_imageContrastEnhancement_params,
    ImageDehazeParams,
    ImageDehazeResponse,
    ImageDehazeParamMap,
    validate_imageDehaze_params,
    ImageLosslessEnlargementParams,
    ImageLosslessEnlargementResponse,
    ImageLosslessEnlargementParamMap,
    validate_imageLosslessEnlargement_params,
    ImageClarityEnhancementParams,
    ImageClarityEnhancementResponse,
    ImageClarityEnhancementParamMap,
    validate_imageClarityEnhancement_params,
    ImageDistortionCorrectionParams,
    ImageDistortionCorrectionResponse,
    ImageDistortionCorrectionParamMap,
    validate_imageDistortionCorrection_params,
    ImageCompositionAestheticsScoreParams,
    ImageCompositionAestheticsScoreResponse,
    ImageCompositionAestheticsScoreParamMap,
    validate_imageCompositionAestheticsScore_params,
    ImageExposureRatingParams,
    ImageExposureRatingResponse,
    ImageExposureRatingParamMap,
    validate_imageExposureRating_params,
    ImageAINailArtParams,
    ImageAINailArtResponse,
    ImageAINailArtParamMap,
    validate_imageAINailArt_params,
    ImageAINailArtProParams,
    ImageAINailArtProResponse,
    ImageAINailArtProParamMap,
    validate_imageAINailArtPro_params,
    ImageAIPhotographyParams,
    ImageAIPhotographyResponse,
    ImageAIPhotographyParamMap,
    validate_imageAIPhotography_params,
    ImageAIEmojiGeneratorParams,
    ImageAIEmojiGeneratorResponse,
    ImageAIEmojiGeneratorParamMap,
    validate_imageAIEmojiGenerator_params,
    ImagePhotoToColoringPageParams,
    ImagePhotoToColoringPageResponse,
    ImagePhotoToColoringPageParamMap,
    validate_imagePhotoToColoringPage_params,
    ImageAIFlowerWallpaperParams,
    ImageAIFlowerWallpaperResponse,
    ImageAIFlowerWallpaperParamMap,
    validate_imageAIFlowerWallpaper_params
)

class ImageAPI:
    def __init__(self, requester: Requester):
        self._requester = requester

    async def imageQueryingAsyncTaskResults(self, params: ImageQueryingAsyncTaskResultsParams) -> ImageQueryingAsyncTaskResultsResponse:
        validate_imageQueryingAsyncTaskResults_params(params)
        mapped = map_params(params, ImageQueryingAsyncTaskResultsParamMap)
        return await self._requester.request('GET', '/api/image/asyn-task-results', query=mapped)

    async def imageAiImageExtender(self, params: ImageAiImageExtenderParams) -> ImageAiImageExtenderResponse:
        validate_imageAiImageExtender_params(params)
        mapped = map_params(params, ImageAiImageExtenderParamMap)
        return await self._requester.request('POST', '/api/image/editing/ai-image-extender', body=mapped, multipart=True)

    async def imageAiObjectReplacer(self, params: ImageAiObjectReplacerParams) -> ImageAiObjectReplacerResponse:
        validate_imageAiObjectReplacer_params(params)
        mapped = map_params(params, ImageAiObjectReplacerParamMap)
        return await self._requester.request('POST', '/api/image/editing/ai-object-replacer', body=mapped, multipart=True)

    async def imageAIImageCropping(self, params: ImageAIImageCroppingParams) -> ImageAIImageCroppingResponse:
        validate_imageAIImageCropping_params(params)
        mapped = map_params(params, ImageAIImageCroppingParamMap)
        return await self._requester.request('POST', '/api/image/editing/image-cropping', body=mapped, multipart=True)

    async def imageErasure(self, params: ImageErasureParams) -> ImageErasureResponse:
        validate_imageErasure_params(params)
        mapped = map_params(params, ImageErasureParamMap)
        return await self._requester.request('POST', '/api/image/editing/image-erase', body=mapped, multipart=True)

    async def imageInvisibleImageWatermark(self, params: ImageInvisibleImageWatermarkParams) -> ImageInvisibleImageWatermarkResponse:
        validate_imageInvisibleImageWatermark_params(params)
        mapped = map_params(params, ImageInvisibleImageWatermarkParamMap)
        return await self._requester.request('POST', '/api/image/editing/image-invisible-image-watermark', body=mapped, multipart=True)

    async def imageInvisibleTextWatermark(self, params: ImageInvisibleTextWatermarkParams) -> ImageInvisibleTextWatermarkResponse:
        validate_imageInvisibleTextWatermark_params(params)
        mapped = map_params(params, ImageInvisibleTextWatermarkParamMap)
        return await self._requester.request('POST', '/api/image/editing/image-invisible-text-watermarking', body=mapped, multipart=True)

    async def imageIntelligentComposition(self, params: ImageIntelligentCompositionParams) -> ImageIntelligentCompositionResponse:
        validate_imageIntelligentComposition_params(params)
        mapped = map_params(params, ImageIntelligentCompositionParamMap)
        return await self._requester.request('POST', '/api/image/editing/intelligent-composition', body=mapped, multipart=True)

    async def imagePhotoEditing(self, params: ImagePhotoEditingParams) -> ImagePhotoEditingResponse:
        validate_imagePhotoEditing_params(params)
        mapped = map_params(params, ImagePhotoEditingParamMap)
        return await self._requester.request('POST', '/api/image/editing/photo-retouching', body=mapped, multipart=True)

    async def imageRemoveObjects(self, params: ImageRemoveObjectsParams) -> ImageRemoveObjectsResponse:
        validate_imageRemoveObjects_params(params)
        mapped = map_params(params, ImageRemoveObjectsParamMap)
        return await self._requester.request('POST', '/api/image/editing/remove-objects', body=mapped, multipart=True)

    async def remove_objects(self, params: ImageRemoveObjectsParams) -> ImageRemoveObjectsResponse:
        return await self.imageRemoveObjects(params)

    async def imageRemoveObjectsAdvanced(self, params: ImageRemoveObjectsAdvancedParams) -> ImageRemoveObjectsAdvancedResponse:
        validate_imageRemoveObjectsAdvanced_params(params)
        mapped = map_params(params, ImageRemoveObjectsAdvancedParamMap)
        return await self._requester.request('POST', '/api/image/editing/remove-objects-advanced', body=mapped, multipart=True)

    async def imageRemoveObjectsPro(self, params: ImageRemoveObjectsProParams) -> ImageRemoveObjectsProResponse:
        validate_imageRemoveObjectsPro_params(params)
        mapped = map_params(params, ImageRemoveObjectsProParamMap)
        return await self._requester.request('POST', '/api/image/editing/remove-objects-pro', body=mapped, multipart=True)

    async def imageAICartoonGenerator(self, params: ImageAICartoonGeneratorParams) -> ImageAICartoonGeneratorResponse:
        validate_imageAICartoonGenerator_params(params)
        mapped = map_params(params, ImageAICartoonGeneratorParamMap)
        return await self._requester.request('POST', '/api/image/effects/ai-anime-generator', body=mapped, multipart=True)

    async def imageColoring(self, params: ImageColoringParams) -> ImageColoringResponse:
        validate_imageColoring_params(params)
        mapped = map_params(params, ImageColoringParamMap)
        return await self._requester.request('POST', '/api/image/effects/image-colorization', body=mapped, multipart=True)

    async def imageStyleTransfer(self, params: ImageStyleTransferParams) -> ImageStyleTransferResponse:
        validate_imageStyleTransfer_params(params)
        mapped = map_params(params, ImageStyleTransferParamMap)
        return await self._requester.request('POST', '/api/image/effects/image-style-conversion', body=mapped, multipart=True)

    async def imageStyleMigration(self, params: ImageStyleMigrationParams) -> ImageStyleMigrationResponse:
        validate_imageStyleMigration_params(params)
        mapped = map_params(params, ImageStyleMigrationParamMap)
        return await self._requester.request('POST', '/api/image/effects/image-style-migration', body=mapped, multipart=True)

    async def imageColorEnhancement(self, params: ImageColorEnhancementParams) -> ImageColorEnhancementResponse:
        validate_imageColorEnhancement_params(params)
        mapped = map_params(params, ImageColorEnhancementParamMap)
        return await self._requester.request('POST', '/api/image/enhance/image-color-enhancement', body=mapped, multipart=True)

    async def imageContrastEnhancement(self, params: ImageContrastEnhancementParams) -> ImageContrastEnhancementResponse:
        validate_imageContrastEnhancement_params(params)
        mapped = map_params(params, ImageContrastEnhancementParamMap)
        return await self._requester.request('POST', '/api/image/enhance/image-contrast-enhancement', body=mapped, multipart=True)

    async def imageDehaze(self, params: ImageDehazeParams) -> ImageDehazeResponse:
        validate_imageDehaze_params(params)
        mapped = map_params(params, ImageDehazeParamMap)
        return await self._requester.request('POST', '/api/image/enhance/image-defogging', body=mapped, multipart=True)

    async def imageLosslessEnlargement(self, params: ImageLosslessEnlargementParams) -> ImageLosslessEnlargementResponse:
        validate_imageLosslessEnlargement_params(params)
        mapped = map_params(params, ImageLosslessEnlargementParamMap)
        return await self._requester.request('POST', '/api/image/enhance/image-lossless-enlargement', body=mapped, multipart=True)

    async def upscale(self, params: ImageLosslessEnlargementParams) -> ImageLosslessEnlargementResponse:
        return await self.imageLosslessEnlargement(params)

    async def imageClarityEnhancement(self, params: ImageClarityEnhancementParams) -> ImageClarityEnhancementResponse:
        validate_imageClarityEnhancement_params(params)
        mapped = map_params(params, ImageClarityEnhancementParamMap)
        return await self._requester.request('POST', '/api/image/enhance/image-sharpness-enhancement', body=mapped, multipart=True)

    async def imageDistortionCorrection(self, params: ImageDistortionCorrectionParams) -> ImageDistortionCorrectionResponse:
        validate_imageDistortionCorrection_params(params)
        mapped = map_params(params, ImageDistortionCorrectionParamMap)
        return await self._requester.request('POST', '/api/image/enhance/stretch-image-recovery', body=mapped, multipart=True)

    async def imageCompositionAestheticsScore(self, params: ImageCompositionAestheticsScoreParams) -> ImageCompositionAestheticsScoreResponse:
        validate_imageCompositionAestheticsScore_params(params)
        mapped = map_params(params, ImageCompositionAestheticsScoreParamMap)
        return await self._requester.request('POST', '/api/image/rating/image-composition-aesthetics-scoring', body=mapped, multipart=True)

    async def imageExposureRating(self, params: ImageExposureRatingParams) -> ImageExposureRatingResponse:
        validate_imageExposureRating_params(params)
        mapped = map_params(params, ImageExposureRatingParamMap)
        return await self._requester.request('POST', '/api/image/rating/image-exposure-score', body=mapped, multipart=True)

    async def imageAINailArt(self, params: ImageAINailArtParams) -> ImageAINailArtResponse:
        validate_imageAINailArt_params(params)
        mapped = map_params(params, ImageAINailArtParamMap)
        return await self._requester.request('POST', '/api/image/editing/ai-nail-art', body=mapped, multipart=True)

    async def imageAINailArtPro(self, params: ImageAINailArtProParams) -> ImageAINailArtProResponse:
        validate_imageAINailArtPro_params(params)
        mapped = map_params(params, ImageAINailArtProParamMap)
        return await self._requester.request('POST', '/api/image/editing/ai-nail-art-pro', body=mapped, multipart=True)

    async def imageAIPhotography(self, params: ImageAIPhotographyParams) -> ImageAIPhotographyResponse:
        validate_imageAIPhotography_params(params)
        mapped = map_params(params, ImageAIPhotographyParamMap)
        return await self._requester.request('POST', '/api/image/effects/ai-photography', body=mapped, multipart=True)

    async def imageAIEmojiGenerator(self, params: ImageAIEmojiGeneratorParams) -> ImageAIEmojiGeneratorResponse:
        validate_imageAIEmojiGenerator_params(params)
        mapped = map_params(params, ImageAIEmojiGeneratorParamMap)
        return await self._requester.request('POST', '/api/image/effects/photo-to-emoji-grid', body=mapped, multipart=True)

    async def imagePhotoToColoringPage(self, params: ImagePhotoToColoringPageParams) -> ImagePhotoToColoringPageResponse:
        validate_imagePhotoToColoringPage_params(params)
        mapped = map_params(params, ImagePhotoToColoringPageParamMap)
        return await self._requester.request('POST', '/api/image/effects/photo-to-line-art', body=mapped, multipart=True)

    async def imageAIFlowerWallpaper(self, params: ImageAIFlowerWallpaperParams) -> ImageAIFlowerWallpaperResponse:
        validate_imageAIFlowerWallpaper_params(params)
        mapped = map_params(params, ImageAIFlowerWallpaperParamMap)
        return await self._requester.request('POST', '/api/image/generation/ai-flower-wallpaper', body=mapped, multipart=True)

File: packages/node/src/image/image.ts
import { Requester } from "../http";
import { mapParams } from "../utils";
import {
  ImageQueryingAsyncTaskResultsParams,
  ImageQueryingAsyncTaskResultsResponse,
  ImageQueryingAsyncTaskResultsResponseData,
  validateImageQueryingAsyncTaskResultsParams,
  ImageQueryingAsyncTaskResultsParamMap,
  ImageAiImageExtenderParams,
  ImageAiImageExtenderResponse,
  ImageAiImageExtenderResponseData,
  validateImageAiImageExtenderParams,
  ImageAiImageExtenderParamMap,
  ImageAiObjectReplacerParams,
  ImageAiObjectReplacerResponse,
  ImageAiObjectReplacerResponseData,
  validateImageAiObjectReplacerParams,
  ImageAiObjectReplacerParamMap,
  ImageAIImageCroppingParams,
  ImageAIImageCroppingResponse,
  ImageAIImageCroppingResponseData,
  validateImageAIImageCroppingParams,
  ImageAIImageCroppingParamMap,
  ImageErasureParams,
  ImageErasureResponse,
  ImageErasureResponseData,
  validateImageErasureParams,
  ImageErasureParamMap,
  ImageInvisibleImageWatermarkParams,
  ImageInvisibleImageWatermarkResponse,
  ImageInvisibleImageWatermarkResponseData,
  validateImageInvisibleImageWatermarkParams,
  ImageInvisibleImageWatermarkParamMap,
  ImageInvisibleTextWatermarkParams,
  ImageInvisibleTextWatermarkResponse,
  ImageInvisibleTextWatermarkResponseData,
  validateImageInvisibleTextWatermarkParams,
  ImageInvisibleTextWatermarkParamMap,
  ImageIntelligentCompositionParams,
  ImageIntelligentCompositionResponse,
  ImageIntelligentCompositionResponseData,
  validateImageIntelligentCompositionParams,
  ImageIntelligentCompositionParamMap,
  ImagePhotoEditingParams,
  ImagePhotoEditingResponse,
  ImagePhotoEditingResponseData,
  validateImagePhotoEditingParams,
  ImagePhotoEditingParamMap,
  ImageRemoveObjectsParams,
  ImageRemoveObjectsResponse,
  ImageRemoveObjectsResponseData,
  validateImageRemoveObjectsParams,
  ImageRemoveObjectsParamMap,
  ImageRemoveObjectsAdvancedParams,
  ImageRemoveObjectsAdvancedResponse,
  ImageRemoveObjectsAdvancedResponseData,
  validateImageRemoveObjectsAdvancedParams,
  ImageRemoveObjectsAdvancedParamMap,
  ImageRemoveObjectsProParams,
  ImageRemoveObjectsProResponse,
  ImageRemoveObjectsProResponseData,
  validateImageRemoveObjectsProParams,
  ImageRemoveObjectsProParamMap,
  ImageAICartoonGeneratorParams,
  ImageAICartoonGeneratorResponse,
  ImageAICartoonGeneratorResponseData,
  validateImageAICartoonGeneratorParams,
  ImageAICartoonGeneratorParamMap,
  ImageColoringParams,
  ImageColoringResponse,
  ImageColoringResponseData,
  validateImageColoringParams,
  ImageColoringParamMap,
  ImageStyleTransferParams,
  ImageStyleTransferResponse,
  ImageStyleTransferResponseData,
  validateImageStyleTransferParams,
  ImageStyleTransferParamMap,
  ImageStyleMigrationParams,
  ImageStyleMigrationResponse,
  ImageStyleMigrationResponseData,
  validateImageStyleMigrationParams,
  ImageStyleMigrationParamMap,
  ImageColorEnhancementParams,
  ImageColorEnhancementResponse,
  ImageColorEnhancementResponseData,
  validateImageColorEnhancementParams,
  ImageColorEnhancementParamMap,
  ImageContrastEnhancementParams,
  ImageContrastEnhancementResponse,
  ImageContrastEnhancementResponseData,
  validateImageContrastEnhancementParams,
  ImageContrastEnhancementParamMap,
  ImageDehazeParams,
  ImageDehazeResponse,
  ImageDehazeResponseData,
  validateImageDehazeParams,
  ImageDehazeParamMap,
  ImageLosslessEnlargementParams,
  ImageLosslessEnlargementResponse,
  ImageLosslessEnlargementResponseData,
  validateImageLosslessEnlargementParams,
  ImageLosslessEnlargementParamMap,
  ImageClarityEnhancementParams,
  ImageClarityEnhancementResponse,
  ImageClarityEnhancementResponseData,
  validateImageClarityEnhancementParams,
  ImageClarityEnhancementParamMap,
  ImageDistortionCorrectionParams,
  ImageDistortionCorrectionResponse,
  ImageDistortionCorrectionResponseData,
  validateImageDistortionCorrectionParams,
  ImageDistortionCorrectionParamMap,
  ImageCompositionAestheticsScoreParams,
  ImageCompositionAestheticsScoreResponse,
  ImageCompositionAestheticsScoreResponseData,
  validateImageCompositionAestheticsScoreParams,
  ImageCompositionAestheticsScoreParamMap,
  ImageExposureRatingParams,
  ImageExposureRatingResponse,
  ImageExposureRatingResponseData,
  validateImageExposureRatingParams,
  ImageExposureRatingParamMap,
  ImageAINailArtParams,
  ImageAINailArtResponse,
  ImageAINailArtResponseData,
  validateImageAINailArtParams,
  ImageAINailArtParamMap,
  ImageAINailArtProParams,
  ImageAINailArtProResponse,
  ImageAINailArtProResponseData,
  validateImageAINailArtProParams,
  ImageAINailArtProParamMap,
  ImageAIPhotographyParams,
  ImageAIPhotographyResponse,
  ImageAIPhotographyResponseData,
  validateImageAIPhotographyParams,
  ImageAIPhotographyParamMap,
  ImageAIEmojiGeneratorParams,
  ImageAIEmojiGeneratorResponse,
  ImageAIEmojiGeneratorResponseData,
  validateImageAIEmojiGeneratorParams,
  ImageAIEmojiGeneratorParamMap,
  ImagePhotoToColoringPageParams,
  ImagePhotoToColoringPageResponse,
  ImagePhotoToColoringPageResponseData,
  validateImagePhotoToColoringPageParams,
  ImagePhotoToColoringPageParamMap,
  ImageAIFlowerWallpaperParams,
  ImageAIFlowerWallpaperResponse,
  ImageAIFlowerWallpaperResponseData,
  validateImageAIFlowerWallpaperParams,
  ImageAIFlowerWallpaperParamMap
} from "../generated/endpoints";

export class ImageAPI {
  constructor(private request: Requester) {}

  async imageQueryingAsyncTaskResults(params: ImageQueryingAsyncTaskResultsParams): Promise<ImageQueryingAsyncTaskResultsResponse> {
    validateImageQueryingAsyncTaskResultsParams(params);
    const mapped = mapParams(params, ImageQueryingAsyncTaskResultsParamMap);
    return this.request.request<ImageQueryingAsyncTaskResultsResponseData>("GET", "/api/image/asyn-task-results", { query: mapped });
  }

  async imageAiImageExtender(params: ImageAiImageExtenderParams): Promise<ImageAiImageExtenderResponse> {
    validateImageAiImageExtenderParams(params);
    const mapped = mapParams(params, ImageAiImageExtenderParamMap);
    return this.request.request<ImageAiImageExtenderResponseData>("POST", "/api/image/editing/ai-image-extender", { body: mapped, multipart: true });
  }

  async imageAiObjectReplacer(params: ImageAiObjectReplacerParams): Promise<ImageAiObjectReplacerResponse> {
    validateImageAiObjectReplacerParams(params);
    const mapped = mapParams(params, ImageAiObjectReplacerParamMap);
    return this.request.request<ImageAiObjectReplacerResponseData>("POST", "/api/image/editing/ai-object-replacer", { body: mapped, multipart: true });
  }

  async imageAIImageCropping(params: ImageAIImageCroppingParams): Promise<ImageAIImageCroppingResponse> {
    validateImageAIImageCroppingParams(params);
    const mapped = mapParams(params, ImageAIImageCroppingParamMap);
    return this.request.request<ImageAIImageCroppingResponseData>("POST", "/api/image/editing/image-cropping", { body: mapped, multipart: true });
  }

  async imageErasure(params: ImageErasureParams): Promise<ImageErasureResponse> {
    validateImageErasureParams(params);
    const mapped = mapParams(params, ImageErasureParamMap);
    return this.request.request<ImageErasureResponseData>("POST", "/api/image/editing/image-erase", { body: mapped, multipart: true });
  }

  async imageInvisibleImageWatermark(params: ImageInvisibleImageWatermarkParams): Promise<ImageInvisibleImageWatermarkResponse> {
    validateImageInvisibleImageWatermarkParams(params);
    const mapped = mapParams(params, ImageInvisibleImageWatermarkParamMap);
    return this.request.request<ImageInvisibleImageWatermarkResponseData>("POST", "/api/image/editing/image-invisible-image-watermark", { body: mapped, multipart: true });
  }

  async imageInvisibleTextWatermark(params: ImageInvisibleTextWatermarkParams): Promise<ImageInvisibleTextWatermarkResponse> {
    validateImageInvisibleTextWatermarkParams(params);
    const mapped = mapParams(params, ImageInvisibleTextWatermarkParamMap);
    return this.request.request<ImageInvisibleTextWatermarkResponseData>("POST", "/api/image/editing/image-invisible-text-watermarking", { body: mapped, multipart: true });
  }

  async imageIntelligentComposition(params: ImageIntelligentCompositionParams): Promise<ImageIntelligentCompositionResponse> {
    validateImageIntelligentCompositionParams(params);
    const mapped = mapParams(params, ImageIntelligentCompositionParamMap);
    return this.request.request<ImageIntelligentCompositionResponseData>("POST", "/api/image/editing/intelligent-composition", { body: mapped, multipart: true });
  }

  async imagePhotoEditing(params: ImagePhotoEditingParams): Promise<ImagePhotoEditingResponse> {
    validateImagePhotoEditingParams(params);
    const mapped = mapParams(params, ImagePhotoEditingParamMap);
    return this.request.request<ImagePhotoEditingResponseData>("POST", "/api/image/editing/photo-retouching", { body: mapped, multipart: true });
  }

  async imageRemoveObjects(params: ImageRemoveObjectsParams): Promise<ImageRemoveObjectsResponse> {
    validateImageRemoveObjectsParams(params);
    const mapped = mapParams(params, ImageRemoveObjectsParamMap);
    return this.request.request<ImageRemoveObjectsResponseData>("POST", "/api/image/editing/remove-objects", { body: mapped, multipart: true });
  }

  async removeObjects(params: ImageRemoveObjectsParams): Promise<ImageRemoveObjectsResponse> {
    return this.imageRemoveObjects(params);
  }

  async imageRemoveObjectsAdvanced(params: ImageRemoveObjectsAdvancedParams): Promise<ImageRemoveObjectsAdvancedResponse> {
    validateImageRemoveObjectsAdvancedParams(params);
    const mapped = mapParams(params, ImageRemoveObjectsAdvancedParamMap);
    return this.request.request<ImageRemoveObjectsAdvancedResponseData>("POST", "/api/image/editing/remove-objects-advanced", { body: mapped, multipart: true });
  }

  async imageRemoveObjectsPro(params: ImageRemoveObjectsProParams): Promise<ImageRemoveObjectsProResponse> {
    validateImageRemoveObjectsProParams(params);
    const mapped = mapParams(params, ImageRemoveObjectsProParamMap);
    return this.request.request<ImageRemoveObjectsProResponseData>("POST", "/api/image/editing/remove-objects-pro", { body: mapped, multipart: true });
  }

  async imageAICartoonGenerator(params: ImageAICartoonGeneratorParams): Promise<ImageAICartoonGeneratorResponse> {
    validateImageAICartoonGeneratorParams(params);
    const mapped = mapParams(params, ImageAICartoonGeneratorParamMap);
    return this.request.request<ImageAICartoonGeneratorResponseData>("POST", "/api/image/effects/ai-anime-generator", { body: mapped, multipart: true });
  }

  async imageColoring(params: ImageColoringParams): Promise<ImageColoringResponse> {
    validateImageColoringParams(params);
    const mapped = mapParams(params, ImageColoringParamMap);
    return this.request.request<ImageColoringResponseData>("POST", "/api/image/effects/image-colorization", { body: mapped, multipart: true });
  }

  async imageStyleTransfer(params: ImageStyleTransferParams): Promise<ImageStyleTransferResponse> {
    validateImageStyleTransferParams(params);
    const mapped = mapParams(params, ImageStyleTransferParamMap);
    return this.request.request<ImageStyleTransferResponseData>("POST", "/api/image/effects/image-style-conversion", { body: mapped, multipart: true });
  }

  async imageStyleMigration(params: ImageStyleMigrationParams): Promise<ImageStyleMigrationResponse> {
    validateImageStyleMigrationParams(params);
    const mapped = mapParams(params, ImageStyleMigrationParamMap);
    return this.request.request<ImageStyleMigrationResponseData>("POST", "/api/image/effects/image-style-migration", { body: mapped, multipart: true });
  }

  async imageColorEnhancement(params: ImageColorEnhancementParams): Promise<ImageColorEnhancementResponse> {
    validateImageColorEnhancementParams(params);
    const mapped = mapParams(params, ImageColorEnhancementParamMap);
    return this.request.request<ImageColorEnhancementResponseData>("POST", "/api/image/enhance/image-color-enhancement", { body: mapped, multipart: true });
  }

  async imageContrastEnhancement(params: ImageContrastEnhancementParams): Promise<ImageContrastEnhancementResponse> {
    validateImageContrastEnhancementParams(params);
    const mapped = mapParams(params, ImageContrastEnhancementParamMap);
    return this.request.request<ImageContrastEnhancementResponseData>("POST", "/api/image/enhance/image-contrast-enhancement", { body: mapped, multipart: true });
  }

  async imageDehaze(params: ImageDehazeParams): Promise<ImageDehazeResponse> {
    validateImageDehazeParams(params);
    const mapped = mapParams(params, ImageDehazeParamMap);
    return this.request.request<ImageDehazeResponseData>("POST", "/api/image/enhance/image-defogging", { body: mapped, multipart: true });
  }

  async imageLosslessEnlargement(params: ImageLosslessEnlargementParams): Promise<ImageLosslessEnlargementResponse> {
    validateImageLosslessEnlargementParams(params);
    const mapped = mapParams(params, ImageLosslessEnlargementParamMap);
    return this.request.request<ImageLosslessEnlargementResponseData>("POST", "/api/image/enhance/image-lossless-enlargement", { body: mapped, multipart: true });
  }

  async upscale(params: ImageLosslessEnlargementParams): Promise<ImageLosslessEnlargementResponse> {
    return this.imageLosslessEnlargement(params);
  }

  async imageClarityEnhancement(params: ImageClarityEnhancementParams): Promise<ImageClarityEnhancementResponse> {
    validateImageClarityEnhancementParams(params);
    const mapped = mapParams(params, ImageClarityEnhancementParamMap);
    return this.request.request<ImageClarityEnhancementResponseData>("POST", "/api/image/enhance/image-sharpness-enhancement", { body: mapped, multipart: true });
  }

  async imageDistortionCorrection(params: ImageDistortionCorrectionParams): Promise<ImageDistortionCorrectionResponse> {
    validateImageDistortionCorrectionParams(params);
    const mapped = mapParams(params, ImageDistortionCorrectionParamMap);
    return this.request.request<ImageDistortionCorrectionResponseData>("POST", "/api/image/enhance/stretch-image-recovery", { body: mapped, multipart: true });
  }

  async imageCompositionAestheticsScore(params: ImageCompositionAestheticsScoreParams): Promise<ImageCompositionAestheticsScoreResponse> {
    validateImageCompositionAestheticsScoreParams(params);
    const mapped = mapParams(params, ImageCompositionAestheticsScoreParamMap);
    return this.request.request<ImageCompositionAestheticsScoreResponseData>("POST", "/api/image/rating/image-composition-aesthetics-scoring", { body: mapped, multipart: true });
  }

  async imageExposureRating(params: ImageExposureRatingParams): Promise<ImageExposureRatingResponse> {
    validateImageExposureRatingParams(params);
    const mapped = mapParams(params, ImageExposureRatingParamMap);
    return this.request.request<ImageExposureRatingResponseData>("POST", "/api/image/rating/image-exposure-score", { body: mapped, multipart: true });
  }

  async imageAINailArt(params: ImageAINailArtParams): Promise<ImageAINailArtResponse> {
    validateImageAINailArtParams(params);
    const mapped = mapParams(params, ImageAINailArtParamMap);
    return this.request.request<ImageAINailArtResponseData>("POST", "/api/image/editing/ai-nail-art", { body: mapped, multipart: true });
  }

  async imageAINailArtPro(params: ImageAINailArtProParams): Promise<ImageAINailArtProResponse> {
    validateImageAINailArtProParams(params);
    const mapped = mapParams(params, ImageAINailArtProParamMap);
    return this.request.request<ImageAINailArtProResponseData>("POST", "/api/image/editing/ai-nail-art-pro", { body: mapped, multipart: true });
  }

  async imageAIPhotography(params: ImageAIPhotographyParams): Promise<ImageAIPhotographyResponse> {
    validateImageAIPhotographyParams(params);
    const mapped = mapParams(params, ImageAIPhotographyParamMap);
    return this.request.request<ImageAIPhotographyResponseData>("POST", "/api/image/effects/ai-photography", { body: mapped, multipart: true });
  }

  async imageAIEmojiGenerator(params: ImageAIEmojiGeneratorParams): Promise<ImageAIEmojiGeneratorResponse> {
    validateImageAIEmojiGeneratorParams(params);
    const mapped = mapParams(params, ImageAIEmojiGeneratorParamMap);
    return this.request.request<ImageAIEmojiGeneratorResponseData>("POST", "/api/image/effects/photo-to-emoji-grid", { body: mapped, multipart: true });
  }

  async imagePhotoToColoringPage(params: ImagePhotoToColoringPageParams): Promise<ImagePhotoToColoringPageResponse> {
    validateImagePhotoToColoringPageParams(params);
    const mapped = mapParams(params, ImagePhotoToColoringPageParamMap);
    return this.request.request<ImagePhotoToColoringPageResponseData>("POST", "/api/image/effects/photo-to-line-art", { body: mapped, multipart: true });
  }

  async imageAIFlowerWallpaper(params: ImageAIFlowerWallpaperParams): Promise<ImageAIFlowerWallpaperResponse> {
    validateImageAIFlowerWallpaperParams(params);
    const mapped = mapParams(params, ImageAIFlowerWallpaperParamMap);
    return this.request.request<ImageAIFlowerWallpaperResponseData>("POST", "/api/image/generation/ai-flower-wallpaper", { body: mapped, multipart: true });
  }

}
